### ISSUE TITLE
Should I use "runDisposables" in "destroy" or "willDestroy"?

### ISSUE THREAD

**Issue by 2hu12** [ember-lifeline/ember-lifeline#143]
I just find all examples and the mixins use "runDisposables" in "destroy" method rather than "willDestroy", could you please explain the reason?

**Comment by scalvert** [comment]
Hey @2hu12, this is a good question. 

Historically, prior to the latest lifeline refactor, mixins were more heavily used. Within those mixins, lifeline used instance arrays to manage the async state to be disposed. Those arrays were [initialized inside the `init` of the mixins](https://github.com/ember-lifeline/ember-lifeline/blob/v2.0.0/addon/mixins/run.js#L42).

The mixin used to use `willDestroy` to dispose of the async state, but after discussing with @rwjblue we decided that since `destroy` was the semantic inverse of `init` we'd change to using that. This was done in [this commit](https://github.com/ember-lifeline/ember-lifeline/commit/7cac491c5beae7f47ec1cdd57e19453f29ec6d49).

The refactor of lifeline to utilize `WeakMap` to manage the association of async work to objects removed the need for the mixin to initialize those instance arrays, therefore the use of `init` was removed. The call to `runDisposables` in `destroy` remained, and is largely a remnant of the previous structure.

Have you encountered an issue with us using `destroy` vs. `willDestroy`?
